### PR TITLE
WIP 5.0 revamp of documentation

### LINKED ISSUE #39: Update documentation for 5.0

### PR DESCRIPTION
**Pull request by joeljfischer**
Fixes #39 

This pull request fixes existing content and adds new content.

## Summary of Changes
This pull request overhauls the guides for 5.0 compatibility.

## TODOs
- [x] Getting Started
- [x] Displaying Information
- [x] Uploading Files and Graphics
  - [x] Batch file uploads
- [x] Knowing In-Car UI Status
- [x] Adding the Lock Screen
- [x] Getting Vehicle Data 
- [x] Getting In-Car Audio (Rename to in-car microphone audio)
- [x] Setting the Navigation Destination
- [x] Calling a Phone Number
- [x] Mobile Navigation
  - [x] Haptic / Focus Item data
- [x] Developer Tools

## New Sections
- [x] ChangeRegistration
- [x] Updating to v5.0 (from v4.3+ / from v4.2-)
- [x] Remote Control

### CODE REVIEW

File: docs/Getting Started/SDK Configuration/index.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ## SDK Configuration
 ### 1. Get  a SDL Core
-If you do not have a SDL enabled head unit for testing, you should build the [sdl_core project](https://github.com/smartdevicelink/sdl_core). The sdl_core project is an emulator that lets you simulate sending and receiving remote procedure calls between a smartphone app and a SDL Core.
+If you do not have an SDL enabled head unit for testing, [manticore](https://smartdevicelink.com/resources/manticore/) may work for you. Manticore is an emulator that lets you simulate a head unit and test your app against it using TCP connections.
```
Comment (NicoleYarroch) [inline docs/Getting Started/SDK Configuration/index.md]: Manticore (in the link) should be capitalized.

File: docs/Adding the Lock Screen/index.md [hunk 2]
```diff
@@ -1,9 +1,9 @@
 ## Adding the Lock Screen
-The lock screen is a vital part of SmartDeviceLink, as this does not allow the user to use your application while the vehicle is in motion. Prior to SDL 4.3, creating and maintaining the lock screen was up to the developer to do. Now, SDL takes care of the lock screen for you. It still allows you to use your own view controller if you prefer your own look, but still want the recommended logic that SDL provides for free.
+The lock screen is a vital part of SmartDeviceLink, as this does not allow the user to use your application while the vehicle is in motion. SDL takes care of the lock screen for you. It still allows you to use your own view controller if you prefer your own look, but still want the recommended logic that SDL provides for free.
```
Comment (NicoleYarroch) [inline docs/Adding the Lock Screen/index.md]: The lock screen is a vital part of SmartDeviceLink, as **the lock screen prevents** the user **from using** your application while the vehicle is in motion.

File: docs/Getting Started/Adapting to the Head Unit Language/index.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+Adapting to the head unit language requires you to observe the register app interface response and use the `SDLChangeRegistration` RPC to update your app's name into the new language if you support it.
+
+### When to support languages
+Matches == The app's language and the head unit language are the same
+Supported == The app supports the language returned by the head unit
+
+| App Language | Head Unit Language  | What to do |
+| ------------ | ------------------- | ---------- |
+| Matches       | Supported   | Nothing, you already match |
+| Matches       | Unsupported | This is impossible |
```
Comment (brettywhite) [inline docs/Getting Started/Adapting to the Head Unit Language/index.md]: As this is impossible, is it necessary to add in here?

File: docs/Getting Started/Connecting to an Infotainment System/index.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+### Connect with a Vehicle Head Unit or a Technical Development Kit (TDK)
+#### Production
+To connect your iOS device directly to a vehicle head unit or TDK, make sure to implement an iAP (`default`) connection in the `SDLLifecycleConfiguration`. Then connect the iOS device to the head unit or TDK using a USB cord or Bluetooth if the head unit supports it.
+
+!!! NOTE
+Ford TDKs do not currently support IAP over Bluetooth.
+!!!
+
+#### Debugging
+If you are testing with a vehicle head unit or TDK and wish to see debug logs in Xcode while the app is running, you must either use another app called the [relay app](https://github.com/smartdevicelink/relay_app_ios) to help you connect to the device, or you may use [Xcode 9 / iOS 11 wireless debugging](https://developer.apple.com/videos/play/wwdc2017/404/). When using the relay app, make sure to implement a TCP connection, if using iOS 11 wireless debuggin, implement a IAP connection. Please see the [guide](Developer Tools/Relay) for the relay app to learn how to setup the connection between the device, the relay app and your app.
```
Comment (brettywhite) [inline docs/Getting Started/Connecting to an Infotainment System/index.md]: debugging

File: docs/Getting Started/Integration Basics/index.md [hunk 5]
```diff
@@ -203,7 +205,7 @@ If you are using a head unit or TDK, and are using the [relay app](https://githu
 
 
 ### 2. Short app name (optional)
-This is a shortened version of your app name that is substituted when the full app name will not be visible due to character count constraints
+This is a shortened version of your app name that is substituted when the full app name will not be visible due to character count constraints. You will want to amek this as short as possible.
```
Comment (brettywhite) [inline docs/Getting Started/Integration Basics/index.md]: make

File: docs/Mobile Navigation/Displaying Turn Directions/index.md [hunk 6]
```diff
@@ -1,37 +1,11 @@
...
+Currently, to display a Turn by Turn advice, a combination of the `SDLShowConstantTBT` and `SDLAlertManeuver` RPCs must be used. The `SDLShowConstantTBT` RPC involves the data that will be shown on the head unit. The main properties of this object to set are `navigationText1`, `navigationText2`, and `turnIcon`. A best practice for navigation applications is to use the `navigationText1` as the advice to give (Turn Right) and `navigationText2` to provide the distance to that advice (3 mi). When an `SDLAlertManeuver` is sent, you may also include accompanying text that you would like the head unit to speak when an advice is displayed on screen (e.g. In 3 miles turn right.).
```
Comment (brettywhite) [inline docs/Mobile Navigation/Displaying Turn Directions/index.md]: Currently, to display a Turn by Turn direction,

File: docs/Mobile Navigation/Displaying Turn Directions/index.md [hunk 7]
```diff
@@ -45,52 +19,78 @@ turnByTurn.turnIcon = turnIcon;
...
+            print("Error sending AlertManeuver.")
+            return
+        }
+    }
+}
 ```
 
+> Remember when sending a SDLImage, that the image must first be uploaded to the head unit with the FileManager.
+
+### Clearing the Turn by Turn Advice
```
Comment (brettywhite) [inline docs/Mobile Navigation/Displaying Turn Directions/index.md]: If we are switching to using "direction" as we did above, it should remain consistent

File: docs/Mobile Navigation/Introduction/index.md [hunk 8]
```diff
@@ -2,44 +2,37 @@
...
+The first difference for a navigation app is the `appHMIType` of `SDLAppHMITypeNavigation` that has to be set in the `SDLLifecycleConfiguration`. Navigation apps are also non-media apps.
 
-The second difference is a property called `securityManagers` that needs to be set in the `SDLLifecycleConfiguration` if connecting to a version of Core that requires secure video & audio streaming. This property requires an array of classes of Security Managers, which will conform to the `SDLSecurityType` protocol. These security libraries are provided by the OEMs themselves, and will only work for that OEM. There is not a general catch-all security library.
+The second difference is that an `SDLStreamingMediaConfiguration` must be created and passed to the `SDLConfiguration`. A property called `securityManagers` must be set if connecting to a version of Core that requires secure video & audio streaming. This property requires an array of classes of Security Managers, which will conform to the `SDLSecurityType` protocol. These security libraries are provided by the OEMs themselves, and will only work for that OEM. There is not a general catch-all security library.
```
Comment (brettywhite) [inline docs/Mobile Navigation/Introduction/index.md]: The second difference is that a `SDLStreamingMediaConfiguration`

File: docs/Uploading Files and Graphics/index.md [hunk 9]
```diff
@@ -96,11 +96,52 @@ sdlManager.fileManager.uploadFile(file) { (success, bytesAvailable, error) in
...
+}) { (error) in
+    // All files have finished uploading or errored
+    // The userInfo of the error will contain type [fileName: error]
+    <#code#>
+}
+```
 
 ### File Persistance
-`SDLFile`, and it's subclass `SDLArtwork`, support uploading persistant images, i.e. images that do not become deleted when your application disconnects. Persistance should be used for images relating to your UI, and not for dynamic aspects, such as Album Artwork.
+`SDLFile`, and it's subclass `SDLArtwork`, support uploading persistant files, i.e. images that do not become deleted the car turns off. Persistance should be used for images relating to your UI, such as soft button images, and not for dynamic aspects, such as Album Artwork.
```
Comment (brettywhite) [inline docs/Uploading Files and Graphics/index.md]: i.e. images that do not become deleted when the car turns off

Comment (theresalech) [inline docs/Uploading Files and Graphics/index.md]: "its"

Also, "...images that do not become deleted when the car turns off." (add "when")

File: docs/Uploading Files and Graphics/index.md [hunk 10]
```diff
@@ -113,11 +154,11 @@ file.persistent = true
 ```
 
 !!! note
-Be aware that persistance will not work if space on the head unit is limited. `SDLFileManager` will always handle uploading images if they are non-existant. 
+Be aware that persistance will not work if space on the head unit is limited. `SDLFileManager` will always handle uploading images if they are non-existant.
```
Comment (theresalech) [inline docs/Uploading Files and Graphics/index.md]: "non-existent."

File: docs/Getting Started/Adapting to the Head Unit Language/index.md (hunk 3 above)
Comment (joeljfischer) [reply docs/Getting Started/Adapting to the Head Unit Language/index.md]: *I* know it's impossible, and *you* know it's impossible...but do *they* know it's impossible?

File: docs/Uploading Files and Graphics/index.md (hunk 9 above)
Comment (joeljfischer) [reply docs/Uploading Files and Graphics/index.md]: Weird, in the text, when is already there.

**Pull request merged**